### PR TITLE
Add new FFI functions rnp_backend_string() and rnp_backend_version().

### LINKED ISSUE #1597: FFI: add function to retrieve used backend and it's version.
## Description

Interface could be as following:
- `RNP_API const char *rnp_backend_string()`
- `RNP_API rnp_result_t rnp_backend_version(char **version)`

### PR DESCRIPTION
**Pull request by antonsviridenko**
Closes #1597

### CODE REVIEW

**Comment by antonsviridenko** [comment]
```
  25/228 Test  #15: rnp_tests.s2k_iteration_tuning ................................................***Failed    0.57 sec
Note: Google Test filter = rnp_tests.s2k_iteration_tuning
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from rnp_tests
[ RUN      ] rnp_tests.s2k_iteration_tuning
/home/runner/work/rnp/rnp/src/tests/cipher.cpp:647: Failure
Expected: ((static_cast<double>(iters_100) / iters_10)) > ((6)), actual: 5.7931 vs 6
[  FAILED  ] rnp_tests.s2k_iteration_tuning (551 ms)
[----------] 1 test from rnp_tests (551 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (552 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] rnp_tests.s2k_iteration_tuning

 1 FAILED TEST
```
not sure if related to my changes

**Comment by antonsviridenko** [comment]
ping @ni4

**Comment by ni4** [comment]
@antonsviridenko Sorry, believed I answered here. That test fails occasionally due to lack of precision in time calculations. Review comes shortly...

File: src/lib/crypto/backend_version.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+namespace rnp {
+
+const char *
+backend_string()
+{
+#if defined(CRYPTO_BACKEND_BOTAN)
+    return "Botan";
+#elif defined(CRYPTO_BACKEND_OPENSSL)
+    return "OpenSSL";
+#endif
```
Comment (ni4) [inline src/lib/crypto/backend_version.cpp:45]: Add `#else #error "Unknown backend"` here?

File: src/lib/crypto/backend_version.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+const char *
+backend_version()
+{
+#if defined(CRYPTO_BACKEND_BOTAN)
+    return Botan::short_version_cstr();
+#elif defined(CRYPTO_BACKEND_OPENSSL)
+    return OpenSSL_version(OPENSSL_VERSION);
```
Comment (ni4) [inline src/lib/crypto/backend_version.cpp:54]: Same as the previous comment.

**Review by ni4 — APPROVED** [review]
LGTM, except two minor issues. Feel free to merge once those are addressed.

**Pull request merged**